source/rust_verify/src/verifier.rs

```diff
@@ -934,6 +958,55 @@ impl Verifier {
                 .join("\n");
                 return Err(error(msg));
             }
+
+            // substring match
+            if verify_function == clean_verify_function {
+                exact_match = filtered_functions.iter().any(|f| f == &verify_function);
+                if filtered_functions.len() > 1 && !exact_match {
+                    filtered_functions.sort();
+                    let msg = vec![
+                        format!(
+                            "more than one match found for --verify-function {verify_function}, consider using wildcard *{verify_function}* to verify all matched results,"
+                        ),
+                        format!(
+                            "or specify a unique substring for the desired function, matched results are:"
+                        ),
+                    ].into_iter()
+                    .chain(filtered_functions.iter().map(|f| format!("  - {f}")))
+                    .collect::<Vec<String>>()
+                    .join("\n");
+                    return Err(error(msg));
+                }
+            } else {
+                // wildcard match
+                let left_wildcard = verify_function.starts_with("*");
+                let right_wildcard = verify_function.ends_with("*");
+                let mut wildcard_mismatch = false;
+
+                if left_wildcard && !right_wildcard {
+                    wildcard_mismatch =
+                        !filtered_functions.iter().any(|f| f.ends_with(clean_verify_function));
+                } else if !left_wildcard && right_wildcard {
+                    wildcard_mismatch =
+                        !filtered_functions.iter().any(|f| f.starts_with(clean_verify_function));
+                }
```

Comment:
Alternative way of writing this code that eliminates the `let mut`
```suggestion
                let wildcard_mismatch = match (verify_function.starts_with("*"), verify_function.ends_with("*")) {
                  (true, false) => !filtered_functions.iter().any(|f| f.ends_with(clean_verify_function)),
                  (false, true) => !filtered_functions.iter().any(|f| f.starts_with(clean_verify_function)),
                  _ => false,
                };
```

You can also instead do it as a `let wildcard_mismatch = if ...` if you prefer that